src/vcpkg/base/files.cpp

```diff
@@ -2653,8 +2701,48 @@ namespace vcpkg
                                std::error_code& ec) override
         {
 #if defined(_WIN32)
-            return stdfs::copy_file(
-                to_stdfs_path(source), to_stdfs_path(destination), convert_copy_options(options), ec);
+            auto wide_source = Strings::to_utf16(source.native());
```

Comment:
@dg0yt Please review whether this completely-win32-native version is OK and let me know; otherwise, just rebase the commit adding it off.

Reply:
(I followed the same strategy as when I implemented `std::filesystem` in the STL: https://github.com/microsoft/STL/blob/178b8406a5647507e759442960f17cd33045cdf5/stl/src/filesystem.cpp#L345 )